hazelcast/src/main/java/com/hazelcast/map/impl/query/AggregationResultProcessor.java

```diff
@@ -39,6 +43,7 @@ public AggregationResult populateResult(Query query, long resultLimit, Collectio
 
     @Override
     public AggregationResult populateResult(Query query, long resultLimit) {
-        return new AggregationResult(query.getAggregator());
+        Aggregator resultAggregator = serializationService.toObject(serializationService.toData(query.getAggregator()));
```

Comment:
Can you elaborate which a `toObject(toData())` is needed? Can we limit this to a special case or do we have to do this all the time? It looks a bit expensive and just seems to be necessary in some use cases.